lib/gems/pending/util/vmdb-logger.rb

```diff
@@ -108,9 +108,13 @@ def self.log_hashes(logger, h, options = {})
     filter = Array(options[:filter]).flatten.compact.map(&:to_s) << "password"
     filter.uniq!
 
-    YAML.dump(h).split("\n").each do |l|
+    values = YAML.dump(h).gsub(MiqPassword::REGEXP) { |m| m.gsub($2, "FILTERED") }
+    values.split("\n").each do |l|
       next if l[0...3] == '---'
-      logger.send(level, "  #{l}") unless filter.any? { |f| l.include?(f) }
+      if key = filter.detect { |f| l.include?(f) }
+        l.gsub!(/#{key}.*: (.+)/) { |m| m.gsub!($1, "[FILTERED]") }
```

Comment:
Instead of doing string manipulation here, why not just dup the hash passed in and alter the value before doing the `YAML.dump`? That seems like it would be much more reliable than this regex dance.